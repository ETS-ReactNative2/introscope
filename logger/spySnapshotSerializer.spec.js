const { proxySpy } = require('./proxySpy');
const { spySnapshotSerializer } = require('./spySnapshotSerializer');

global.expect.addSnapshotSerializer(spySnapshotSerializer);

const newSpy = v => proxySpy(() => {}, 'spy', v);

describe('spySnapshotSerializer', () => {
    it('naked spy', () => {
        expect(newSpy([])).toMatchSnapshot();
    });

    it('spied arrays', () => {
        expect([
            newSpy([1, 2, 3]),
            newSpy([4, 5, 6]),
            newSpy([7, 8, 9]),
        ]).toMatchSnapshot();
    });

    it('spied objects', () => {
        expect({
            a: newSpy({ a: 'a' }),
            b: newSpy({ b: 'b' }),
            c: newSpy({ c: 'c' }),
        }).toMatchSnapshot();
    });

    it('spied function', () => {
        expect({
            fun: newSpy(() => {}),
        }).toMatchSnapshot();
    });

    it('spied primitives', () => {
        expect([
            newSpy(true),
            newSpy(1),
            newSpy('foo'),
            newSpy(Symbol('bar')),
        ]).toMatchSnapshot();
    });

    it('keeps other objects', () => {
        expect(['str', 5, [1, 2, 3], () => {}]).toMatchSnapshot();
    });
});
