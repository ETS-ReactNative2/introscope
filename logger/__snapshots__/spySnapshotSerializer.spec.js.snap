// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spySnapshotSerializer keeps other objects 1`] = `
Array [
  "str",
  5,
  Array [
    1,
    2,
    3,
  ],
  [Function],
]
`;

exports[`spySnapshotSerializer naked spy 1`] = `Array []`;

exports[`spySnapshotSerializer spied arrays 1`] = `
Array [
  Array [
    1,
    2,
    3,
  ],
  Array [
    4,
    5,
    6,
  ],
  Array [
    7,
    8,
    9,
  ],
]
`;

exports[`spySnapshotSerializer spied function 1`] = `
Object {
  "fun": [Function],
}
`;

exports[`spySnapshotSerializer spied objects 1`] = `
Object {
  "a": Object {
    "a": "a",
  },
  "b": Object {
    "b": "b",
  },
  "c": Object {
    "c": "c",
  },
}
`;

exports[`spySnapshotSerializer spied primitives 1`] = `
Array [
  true,
  1,
  "foo",
  Symbol(bar),
]
`;
